lib/SparqlXmlParser.ts

```diff
@@ -29,79 +28,121 @@ export class SparqlXmlParser {
    * @return {NodeJS.ReadableStream} A stream of bindings.
    */
   public parseXmlResultsStream(sparqlResponseStream: NodeJS.ReadableStream): NodeJS.ReadableStream {
-    // Collect variables
-    const variables: RDF.Variable[] = [];
-    sparqlResponseStream
-      .pipe(XmlNode({ strict: true, tag: 'variable' }))
-      .on('data', (node: any) => variables.push(this.dataFactory.variable(node.attribs.name)))
-      .on('error', () => { return; }) // Ignore errors, they will emitted in the results
-      .on('finish', () => resultStream.emit('variables', variables));
-
-    // Collect results
-    const resultStream = sparqlResponseStream
-      .pipe(XmlNode({ strict: true, tag: 'result' }))
-      .on('error', (error: Error) => resultStream.emit('error', error))
-      .pipe(new SparqlXmlBindingsTransformer(this));
-
-    // Propagate errors
-    sparqlResponseStream.on('error', (error) => resultStream.emit('error', error));
-
-    return resultStream;
-  }
+    const errorListener = (error: Error) => resultStream.emit('error', error);
+    sparqlResponseStream.on('error', errorListener);
 
-  /**
-   * Convert a SPARQL XML result binding to a bindings object.
-   * @param rawBindings A SPARQL XML result binding.
-   * @return {IBindings} A bindings object.
-   */
-  public parseXmlBindings(rawBindings: any): IBindings {
-    const bindings: IBindings = {};
-    if (rawBindings.children) {
-      const bindingsArray = Array.isArray(rawBindings.children.binding)
-        ? rawBindings.children.binding : [rawBindings.children.binding];
-      for (const binding of bindingsArray) {
-        if (binding.attribs && binding.children) {
-          const key = binding.attribs.name;
-          let term: RDF.Term = null;
-          if (binding.children.bnode) {
-            term = this.dataFactory.blankNode(binding.children.bnode.value);
-          } else if (binding.children.literal) {
-            const value = binding.children.literal.value || "";
-            const attribs = binding.children.literal.attribs;
-            if (attribs && attribs['xml:lang']) {
-              term = this.dataFactory.literal(value, attribs['xml:lang']);
-            } else if (attribs && attribs.datatype) {
-              term = this.dataFactory.literal(value, this.dataFactory.namedNode(attribs.datatype));
-            } else {
-              term = this.dataFactory.literal(value);
-            }
-          } else {
-            term = this.dataFactory.namedNode(binding.children.uri.value);
+    const parser = new SaxesParser({
```

Comment:
I haven't gone through it in detail yet, but it looks like the code has become a lot more complex.
Couldn't the old code be used anymore in its original form for some reason?

Reply:
Yes, I had to move out of `sax-stream` to use `saxes` instead of `sax`.